packages/manager/apps/ips/src/pages/byoip/sections/TokenSection.component.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { OdsText, OdsSpinner } from '@ovhcloud/ods-components/react';
+import { ODS_TEXT_PRESET, ODS_SPINNER_SIZE } from '@ovhcloud/ods-components';
+import { Clipboard } from '@ovh-ux/manager-react-components';
+import { OrderSection } from '@/components/OrderSection/OrderSection.component';
+import { ByoipContext } from '../Byoip.context';
+import { useGetTokens } from '@/data/hooks/useGetTokens';
+
+export const TokenSelectionSection: React.FC = () => {
+  const { t } = useTranslation('byoip');
+  const { selectedRegion } = React.useContext(ByoipContext);
+  const { selectedToken, setSelectedToken } = React.useContext(ByoipContext);
+
+  const { token, isLoading } = useGetTokens({
+    campus: selectedRegion,
+  });
+
+  useEffect(() => {
+    setSelectedToken(token);
+  }, [token]);
```

Comment:
You shouldn't store the token inside the context if it's a derived value from another state. We can just use `useGetTokens` on every other page where we need the token since it's a read only value derived from the selectedRegion state

Reply:
I am storing token on context to display the next section based on its value.

Reply:
You can retrieve the token from react-query like this instead :
```
const qc = useQueryClient();
const data = qc.getQueryData<ApiResponse<string>>(getTokenQueryKey({campus: selectedRegion}));
```